### PR TITLE
[plugin-mobile-app] Add property to specify test execution on real device

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-plugin-mobile-app/src/main/resources/properties/profile/mobile_app/profile.properties [hunk 1]
```diff
@@ -4,3 +4,4 @@ mobile-app=true
 mobile-application.session-scope=story
 mobile-application.swipe.stabilization-duration=PT1S
 mobile-application.swipe.limit=10
+mobile-application.hide-keyboard-using-native-call=true
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/resources/properties/profile/mobile_app/profile.properties:7]: it is not an attribute of mobile application

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/KeyboardActions.java [hunk 2]
```diff
@@ -44,6 +60,16 @@ public KeyboardActions(IWebDriverProvider webDriverProvider)
     public void typeText(WebElement element, String text)
     {
         element.sendKeys(text);
+        // https://github.com/appium/WebDriverAgent/blob/master/WebDriverAgentLib/Commands/FBCustomCommands.m#L107
+        if (genericWebDriverManager.isIOSNativeApp() && !hideKeyboardUsingNativeCall)
+        {
+            Locator keyboardReturnLocator = new Locator(AppiumLocatorType.XPATH, new SearchParameters(
+                    "//XCUIElementTypeKeyboard//XCUIElementTypeButton[@name='Return']", Visibility.VISIBLE, false));
+            WebElement returnButton = searchActions.findElement(keyboardReturnLocator).orElseThrow(
```
[the remark this answers is not in the document]
Comment (uarlouski) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/KeyboardActions.java:68]: then we will have to put body for runnable in order to throw an exception, why not to use api dedicated for this?

File: vividus-plugin-mobile-app/src/main/resources/properties/profile/mobile_app/profile.properties (hunk 1 above)
Comment (uarlouski) [reply vividus-plugin-mobile-app/src/main/resources/properties/profile/mobile_app/profile.properties:7]: ios.hide-keyboard-using-native-call - are you ok with it?

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/KeyboardActions.java [hunk 3]
```diff
@@ -44,6 +60,17 @@ public KeyboardActions(IWebDriverProvider webDriverProvider)
...
+        {
+            Locator keyboardReturnLocator = new Locator(AppiumLocatorType.XPATH, new SearchParameters(
+                    "//XCUIElementTypeKeyboard//XCUIElementTypeButton[@name='Return']", Visibility.VISIBLE, false));
+            searchActions.findElement(keyboardReturnLocator).ifPresentOrElse(touchActions::tap, () ->
+            {
+                throw new IllegalStateException("Unable to find 'Return' button to close the keyboard");
+            });
+            return;
+        }
         webDriverProvider.getUnwrapped(HidesKeyboard.class).hideKeyboard();
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/KeyboardActions.java:74]: if (genericWebDriverManager.isIOSNativeApp() && realDevice)
{
   ...
   ~return;~
}
else
{
        webDriverProvider.getUnwrapped(HidesKeyboard.class).hideKeyboard();
}

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/KeyboardActions.java (hunk 3 above)
Comment (uarlouski) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/KeyboardActions.java]: Fixed, but why not to reduce number of lines by using return?

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**